pkg/sdk/procedures_ext.go

```diff
@@ -108,6 +118,75 @@ func procedureDetailsFromRows(rows []ProcedureDetail) (*ProcedureDetails, error)
 		v.NormalizedArguments = args
 	}
 
+	if v.ExternalAccessIntegrations != nil {
+		if p, err := parseFunctionOrProcedureExternalAccessIntegrations(*v.ExternalAccessIntegrations); err != nil {
+			errs = append(errs, err)
+		} else {
+			v.NormalizedExternalAccessIntegrations = p
+		}
+	} else {
+		v.NormalizedExternalAccessIntegrations = []AccountObjectIdentifier{}
+	}
+
+	if v.Secrets != nil {
+		if p, err := parseFunctionOrProcedureSecrets(*v.Secrets); err != nil {
+			errs = append(errs, err)
+		} else {
+			v.NormalizedSecrets = p
+		}
+	} else {
+		v.NormalizedSecrets = map[string]SchemaObjectIdentifier{}
+	}
+
+	if v.Packages != nil {
+		if p, err := parseFunctionOrProcedurePackages(*v.Packages); err != nil {
+			errs = append(errs, err)
+		} else {
+			// TODO [SNOW-1850370]: merge these and unit test
+			switch strings.ToUpper(v.Language) {
+			case "JAVA", "SCALA":
+				filtered := make([]string, 0)
+				var found bool
+				for _, o := range p {
+					o := strings.TrimSpace(o)
+					if strings.HasPrefix(o, JavaSnowparkPackageString) {
+						v.SnowparkVersion = strings.TrimPrefix(o, JavaSnowparkPackageString)
+						found = true
+					} else {
+						filtered = append(filtered, o)
+					}
+				}
+				v.NormalizedPackages = filtered
+				if !found {
+					errs = append(errs, fmt.Errorf("could not parse package from Snowflake, expected at least snowpark package, got %v", filtered))
+				}
+			case "PYTHON":
+				filtered := make([]string, 0)
+				var found bool
+				for _, o := range p {
+					o := strings.TrimSpace(o)
+					if strings.HasPrefix(o, PythonSnowparkPackageString) {
```

Comment:
Nit: move the switch here, so there's less code duplication.

Reply:
oke